test/system/700-play.bats

```diff
@@ -988,52 +988,94 @@ _EOF
 
-    fname="/tmp/play_kube_wait_$(random_string 6).yaml"
-    echo "
+    _write_test_yaml command=top name=$podname_cmd ctrname=$ctrname_cmd
+    run_podman kube play --annotation "io.podman.annotations.kube.image.volumes.mount/$ctrname_cmd=$imgname1" $TESTYAML
+
+    run_podman run --rm $imgname1 ls -x /
+    run_podman run --rm $imgname2 ls -x /
+    run_podman run --rm $imgname1 ls -x /test1
+    assert "$output" = "a  b  c" "ls /test1 on image"
+    run_out_test1="$output"
+    run_podman exec $podname_cmd-$ctrname_cmd ls -x /test1
+    assert "$output" = "$run_out_test1" "matching ls run/exec volume path test1"
+
+    run_podman run --rm $imgname1 cat /test_same/hello_world
+    assert "$output" = "I am from test1 image"  "cat /test_same/hello_world on image"
+    run_out_test2="$output"
+    run_podman exec $podname_cmd-$ctrname_cmd cat /test_same/hello_world
+    assert "$output" = "$run_out_test2" "matching cat /test_same/hello_world volume path test_same"
+
+    run_podman kube down $TESTYAML
+
+    fname="/$PODMAN_TMPDIR/play_kube_wait_$(random_string 6).yaml"
+    cat >$fname <<EOF
 apiVersion: v1
 kind: Pod
 metadata:
   labels:
     app: test
-  name: $podname
+  name: $podname_file
+  annotations:
+    io.podman.annotations.kube.image.volumes.mount/$ctrname_file: $imgname1;$imgname2
 spec:
   restartPolicy: Never
   containers:
-    - name: $ctrname
+    - name: $ctrname_file
       image: $IMAGE
       command:
       - top
-" > $fname
+    - name: $ctrname_file_not_mounted
+      image: $IMAGE
+      command:
+      - top
+EOF
 
-    run_podman kube play --annotation "io.podman.annotations.kube.image.volumes.mount/$ctrname=$imgname" $fname
+    run_podman kube play $fname
 
-    run_podman run --rm $imgname ls -x /test1
-    assert "$output" = "a  b  c" "ls /test1 on image"
-    run_out_test1="$output"
-    run_podman exec $podname-$ctrname ls -x /test1
-    assert "$output" = "$run_out_test1" "matching ls run/exec volume path test1"
+    run_podman exec "$podname_file-$ctrname_file" ls -x /test1
+    assert "a  b  c" "ls /test1 on image"
 
-    run_podman run --rm $imgname ls -x /test2
-    assert "$output" = "asdf    ejgre   lteghe"  "ls /test2 on image"
-    run_out_test2="$output"
-    run_podman exec $podname-$ctrname ls -x /test2
-    assert "$output" = "$run_out_test2" "matching ls run/exec volume path test2"
+    run_podman exec "$podname_file-$ctrname_file" ls -x /test2
+    assert "asdf    ejgre   lteghe" "ls /test2 on image"
+
+    run_podman exec "$podname_file-$ctrname_file" cat /test_same/hello_world
+    assert "I am from test2 image" "cat /test_same/hello_world inside container"
+
+    run_podman 1 exec "$podname_file-$ctrname_file" touch /test1/readonly
+    assert "$output" =~ "Read-only file system" "image mounted as readonly"
+
+    run_podman exec "$podname_file-$ctrname_file_not_mounted" ls /
+    assert "test" !~ "$output" "No volume should be mounted in $podname_file-$ctrname_file_not_mounted"
```

Comment:
Wrong order, it should be `assert "$output" !~ "test"`. 

Suggestion: just "No volume should be mounted in the no-mount container". There's no reason to include the entire actual container name, that just makes for a hard-to-read error message.